client/grpc-web/src/transports/http/xhr.ts

```diff
@@ -120,6 +126,24 @@ export class MozChunkedArrayBufferXHR extends XHR {
   }
 }
 
+export class ArrayBufferXHR extends XHR {
+  protected configureXhr(): void {
+    this.options.debug && debug("ArrayBufferXHR.configureXhr: setting responseType to 'arraybuffer'");
+    (this.xhr as any).responseType = "arraybuffer";
+  }
+
+  onLoadEvent(): void {
+    const resp = this.xhr.response;
```

Comment:
Does this imply that the `progress` event does not fire? It appears that you are only processing the response data when the XHR has completed (ie: when the `loadend` event is fired).

Reply:
Exactly, this event does not fire. https://github.com/improbable-eng/grpc-web/issues/141#issuecomment-414601236 is a similar solution to what this PR tries to achieve, and this comment also mentions that this event is not being fired.

Reply:
Thanks for clarifying :)